### PR TITLE
Refactored build script and run-script... yyy... script.

### PR DESCRIPTION
**Pull request by MichalCz**
This is a massive refactor of all build scripts.

Two scripts added:

1) `build-all.js` - builds packages:

```
Builds TS and copies results to dist dir
Usage: scripts/build-all.js [options]
    -w,--workspace <name> - workspace filter - default all workspaces
    -d,-dependencies <package> - builds dependencies of a package
    -l,--list - prints list of dirs and exits
    -r,--root <root> - main directory (default is cwd, env: WORKSPACE_ROOT)
    -f,--fast - do not install if package-lock.json exists in dist
    --long-help - for more options
    --outdir - output directory (default <root>/DIST, env: OUT_DIR)
    --ts-config <name> - the name of tsconfig.json file (default is tsconfig.json)
    -D,--dep-types - dependency types ([Dsop]) (default: all)
    --verbose - show verbose output
    --no-build - do not run install in dist, env: NO_BUILD
    --no-dist - do not run copy to dist, env: NO_COPY_DIST
    --no-install - do not run install in dist, env: NO_INSTALL
    --no-distws - do not create package.json in distr, env: NO_WORKSPACE
    --link-packages - hardlink packages via file://, env: LOCAL_PACKAGES
    --flat-packages - copy all packages to one dir, env: FLAT_PACKAGES
    --local-copy - copy license and readme, env: LOCAL_COPY
    --make-public - unprivate the package, env: MAKE_PUBLIC
    --bundle - install locally in the package not workspace
```

2) `scripts/run-scripts.js`

```
Runs scripts in workspaces
Usage: scripts/run-script.js [options] <script-name> [...args]
    -v,--verbose - verbose output
    -s,--scope <path|name> - run in specific package only
    -w,--workspace <name> - workspace filter - default all workspaces
    -d,-dependencies <package> - builds dependencies of a package
    -l,--list - prints list of dirs and exits
    -j,--jobs - how many jobs in parallel (default: cpu count)
    -r,--root <root> - main directory (default is cwd, env: WORKSPACE_ROOT)
    --long-help - for more options
```

The two new scripts are used in package.json but can be used whenever needed.

### CODE REVIEW

**Comment by jan-warchol** [comment]
This removes `lerna`, right? Or is it still used for something in our setup?

**Comment by jan-warchol** [comment]
> This removes `lerna`, right? Or is it still used for something in our setup?

In general, I understand that this is the purpose of this PR - to get rid of lerna?

File: package.json
Comment (jan-warchol) [inline package.json:39]: I'm trying to understand the relationship between these two and `build:*`. `build` happens inside each package, and these moive them to global `dist`, is that right? If so, does this mean that to run STH from compiled JS (e.g. with `yarn start`) I have to run these commands first? (meaning `build:packages` will not be enough?)

Also, should these two be separate? It seems to me that they'll always be ran together anyway.

**Comment by jan-warchol** [comment]
I've found something that looks suspicious: after `yarn build:packages`, python runner source files are not copied to `dist` (and I suppose they should?):

```
$ ll packages/python-runner/dist 
total 24K
drwxrwxr-x 6 jan jan 4,0K maj 30 11:24 ./
drwxrwxr-x 4 jan jan 4,0K maj 30 11:24 ../
drwxrwxr-x 3 jan jan 4,0K maj 30 11:24 pyee/
drwxrwxr-x 2 jan jan 4,0K maj 30 11:24 pyee-8.2.2.dist-info/
drwxrwxr-x 3 jan jan 4,0K maj 30 11:24 scramjet/
drwxrwxr-x 2 jan jan 4,0K maj 30 11:24 scramjet_framework_py-0.10.dist-info/
```

Interestingly, they did end up in global `dist`:
```
$ ll dist/python-runner/
total 92K
drwxrwxr-x  7 jan jan 4,0K maj 30 11:29 ./
drwxrwxr-x 26 jan jan 4,0K maj 30 11:25 ../
drwxrwxr-x  2 jan jan 4,0K maj 30 11:29 __pycache__/
drwxrwxr-x  3 jan jan 4,0K maj 30 11:25 pyee/
drwxrwxr-x  2 jan jan 4,0K maj 30 11:25 pyee-8.2.2.dist-info/
drwxrwxr-x  3 jan jan 4,0K maj 30 11:25 scramjet/
drwxrwxr-x  2 jan jan 4,0K maj 30 11:25 scramjet_framework_py-0.10.dist-info/
-rw-rw-r--  1 jan jan  758 maj 30 11:25 hardcoded_magic_values.py
-rw-rw-r--  1 jan jan  34K maj 30 11:25 LICENSE
-rw-rw-r--  1 jan jan 2,1K maj 30 11:25 logging_setup.py
-rw-rw-r--  1 jan jan  474 maj 30 11:25 package.json
-rw-rw-r--  1 jan jan 3,9K maj 30 11:25 README.md
-rw-rw-r--  1 jan jan  12K maj 30 11:25 runner.py
```

**Comment by jan-warchol** [comment]
Hmm, I see that it doesn't work, but I don't really understand why.

File: packages/types/package.json
Comment (MichalCz) [reply packages/types/package.json]: There's no logic behind this - it's just a change of `postbuild` to `build:only`

**Review by jan-warchol — APPROVED** [review]
I don't feel very confident in giving the approve, as I don't understand well what's happening in the build script. But I think it would take too much time if I tried to.
I get the general idea, so LGTM.

**Review by alicja-gruzdz — APPROVED** [review]

File: package.json
Comment (jan-warchol) [inline package.json:112]: so this is a workspace with everything? And it doesn't collide with others?

Comment (jan-warchol) [reply package.json:112]: whatever, let's not block the pr

**Pull request merged**
